extends Control

## -------------------------------------------------------
## RightPanelUI
##
## √öloha:
## - Zobrazuje z√°kladn√© "survival" resource hodnoty:
##   energia, voda, kysl√≠k, jedlo.
## - ƒå√≠ta priamo z ResourceManageru.
## -------------------------------------------------------

@onready var lbl_energy: Label = $MarginContainer/SurvivalStats/Lbl_Energy
@onready var lbl_water: Label = $MarginContainer/SurvivalStats/Lbl_Water
@onready var lbl_oxygen: Label = $MarginContainer/SurvivalStats/Lbl_Oxygen
@onready var lbl_food: Label = $MarginContainer/SurvivalStats/Lbl_Food


func _ready() -> void:
	## Pri ≈°tarte:
	## - napoj√≠me sa na sign√°l ResourceManageru
	## - inicializujeme v≈°etky texty podƒæa aktu√°lnych hodn√¥t
	if not ResourceManager.resource_changed.is_connected(_on_resource_changed):
		ResourceManager.resource_changed.connect(_on_resource_changed)

	_refresh_all_survival_stats()


func _refresh_all_survival_stats() -> void:
	## Preƒç√≠ta v≈°etky d√¥le≈æit√© resource z ResourceManageru
	## a prep√≠≈°e texty v labeloch.

	var energy: float = ResourceManager.get_amount(ResourceManager.RESOURCE_ENERGY)
	lbl_energy.text = "‚ö° " + str(round(energy))

	var water: float = ResourceManager.get_amount(ResourceManager.RESOURCE_WATER)
	lbl_water.text = "üíß " + str(round(water))

	var oxy: float = ResourceManager.get_amount(ResourceManager.RESOURCE_OXYGEN)
	lbl_oxygen.text = "O‚ÇÇ " + str(round(oxy))

	var food: float = ResourceManager.get_amount(ResourceManager.RESOURCE_FOOD)
	lbl_food.text = "üçΩ " + str(round(food))


func _on_resource_changed(resource_id: StringName, new_amount: float) -> void:
	## Callback zo strany ResourceManageru.
	## Aktualizuje len ten label, ktor√©ho sa zmena t√Ωka.
	match resource_id:
		ResourceManager.RESOURCE_ENERGY:
			lbl_energy.text = "‚ö° " + str(round(new_amount))
		ResourceManager.RESOURCE_WATER:
			lbl_water.text = "üíß " + str(round(new_amount))
		ResourceManager.RESOURCE_OXYGEN:
			lbl_oxygen.text = "O‚ÇÇ " + str(round(new_amount))
		ResourceManager.RESOURCE_FOOD:
			lbl_food.text = "üçΩ " + str(round(new_amount))
		_:
			pass
